server/cmd/code-annotation/handler/assignments.go

```diff
@@ -0,0 +1,74 @@
+package handler
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+
+	"github.com/src-d/code-annotation/server/cmd/code-annotation/serializer"
+	"github.com/src-d/code-annotation/server/model"
+	"github.com/src-d/code-annotation/server/repository"
+)
+
+func HandleGetAssignmentsForUserExperiment(w http.ResponseWriter, r *http.Request) {
+	experimentID, err := strconv.Atoi(chi.URLParam(r, "experimentId"))
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	userID, err := strconv.Atoi("1")
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	assignments, err := repository.GetAssignmentsFor(userID, experimentID)
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	render.Render(w, r, &serializer.ResponseMany{Data: serializer.NewAssignments(assignments)})
+}
+
+func HandleSaveAssignment(w http.ResponseWriter, r *http.Request) {
+	assignmentId, err := strconv.Atoi(chi.URLParam(r, "assignmentId"))
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	var assignment serializer.Assignment
```

Comment:
I usually try to keep structures close to the place where they are used (in the same package and as a private struct) if they are used only once. What is your opinion about it?

Reply:
I think that handlers and serializers should be separated.

Reply:
could you please explain why? How are you going to use serializers beside in handlers? Because in Go usually, you export functions/types if you consider them as your API. 
And it's a good idea to keep your API as small as possible.

Reply:
Packages also help us to choose [good names](https://blog.golang.org/organizing-go-code#TOC_2.).

If we now care about the exposed API, it's because we're taking care of our external users, but this repo does not contain a library to be used for third parties but a repo containing a bunch of things like a web app, tools for DB handling, an API...

If we still think that a curated external API is needed here, we could [move that internal stuff to internal packages](https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU/edit) and keep the benefits of both sides.

Reply:
I still don't like to have a package for only 1 particular consumer. But I can live with it.